Cache blueprint extends

afbora opened the pull request:
## Describe the PR

Blueprint extends are now cached by their name.

However, since the `Blueprint` class is statically called in the tests (_tests/Cms/Blueprints/BlueprintFieldTest.php_), it does not match the data in the cache and the test fails. But in my local test on panel ui works great! Should we refactor the test class (BlueprintFieldTest) or the cache system I wrote was faulty 😇?
I'm creating thr PR as a draft because I'm not sure and waiting your feedback please.

## Related issues

<!-- PR relates to issues in the `kirby` or `idea` repo: -->

- Fixes #2364 

## Ready?

<!-- If you feel like you can help to check off the following tasks, that'd be great. If not, don't worry - we will take care of it. -->

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Fixed code style issues with CS fixer and `composer fix`
- [x] Added in-code documentation (if needed)

afbora committed:
Cache blueprint extends #2364

lukasbestle reviewed (changes requested):
To be honest, I think we should refactor the loading code to avoid having to keep two separate caches (`$loaded` and `$extends`):

- `Blueprint::find()` should always return the blueprint data, not either the data or the file path string. Basically the case where the file exists in the `blueprints` dir should run `Data::read()` and return the array as well.
- With that simplified, the two only callers of that method I can find (`Blueprint::load()` and `Blueprint::extend()`) can be massively simplified to only care for array data. The `if`s for array vs string are no longer required.
- The existing in-memory caching inside `Blueprint::load()` can then be moved to `Blueprint::find()` so that all loaded blueprint files and mixins are cached in the same `$loaded` cache.

lukasbestle committed:
Further cleanup in Blueprints class #2364

lukasbestle committed:
Fix unit test #2364

Caching the blueprint in memory meant that the now overwritten field definition was still cached with the old values; using a different name in this test will circumvent the cache.

bastianallgeier committed:
Merge branch 'develop' into fix/2364-cache-blueprint-extends
